src/admin/resolvers/queries/ApprovedItem.auth.integration.ts

```diff
@@ -0,0 +1,264 @@
+        3
+      );
+
+      await server.stop();
+    });
+
+    it('should throw an error when user does not have the required access', async () => {
+      // Set up auth headers with no valid access group
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_APPROVED_ITEMS,
+      });
+
+      expect(result.data).to.be.null;
+      expect(result.errors).not.to.be.undefined;
+
+      // check if the error we get is access denied error
+      expect(result.errors?.[0].message).to.equal(ACCESS_DENIED_ERROR);
+
+      await server.stop();
+    });
+
+    it('should throw an error when request header groups are undefined', async () => {
+      // Set up auth headers with no valid access group
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: undefined,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_APPROVED_ITEMS,
+      });
+
+      expect(result.data).to.be.null;
+      expect(result.errors).not.to.be.undefined;
+
+      // check if the error we get is access denied error
+      expect(result.errors?.[0].message).to.equal(ACCESS_DENIED_ERROR);
+
+      await server.stop();
+    });
+  });
+
+  describe('getApprovedCuratedCorpusItemByUrl query', () => {
+    it('should get item when user has read-only access', async () => {
+      // Set up auth headers with read-only access
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2,${MozillaAccessGroup.READONLY}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_APPROVED_ITEM_BY_URL,
+        variables: {
+          url: 'https://www.test.com/story-two',
+        },
+      });
+
+      expect(result.data).not.to.be.null;
+      expect(result.errors).to.be.undefined;
+
+      await server.stop();
+    });
+
+    it('should get item when user has only one scheduled surface access', async () => {
+      // Set up auth headers with access to a single Scheduled Surface
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2,${MozillaAccessGroup.NEW_TAB_CURATOR_ENUS}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_APPROVED_ITEM_BY_URL,
+        variables: {
+          url: 'https://www.test.com/story-two',
+        },
+      });
+
+      expect(result.data).not.to.be.null;
```

Comment:
to make this test a little better, we should have at least 2 records, fetch all and ensure that only ones that can be access by this user is returned

Reply:
That's a good idea but for this query, the way the auth works is that if you have access to any one surface, you will get full access. So I will be checking for all the items since all of them will be returned.

Also the reason I omitted checking specific items for this test is because this focused on auth. The other tests in the `ApprovedItem.integration` file cover that part. Omitting the insertion of items in the DB also makes this slightly faster and kinda less redundant in a way 😄